src/chkstat.cpp

```diff
@@ -1033,6 +1007,8 @@ int Chkstat::run() {
     }
 
     if (m_args.system_mode.isSet()) {
+        // this overrides --set
+        m_apply_changes = m_args.only_warn.isSet() ? false : true;
```

Comment:
Yes this is what it boils down to. Previously this read-only mode could also be enabled via sysconfig, or rather, the applying of changes could be disabled. Which is confusing on many levels, because `if CHECK_PERMISSIONS == "true"` then permissions would not only be checked but also be set, while `if CHECK_PERMISSIONS == "false"` then `chkstat` would not continue running at all. Time for a headache pill.